core/ir/aarch64/instr_create.h

```diff
@@ -2514,5 +2520,28 @@
 #define INSTR_CREATE_umull2_vector(dc, Rd, Rm, Rn, width) \
     instr_create_1dst_3src(dc, OP_umull2, Rd, Rm, Rn, width)
 
+/**
+ * Creates a DMB instruction.
+ * \param dc      The void * dcontext used to allocate memory for the instr_t.
+ * \param imm     The integer constant opnd_t operand.
+ */
+#define INSTR_CREATE_dmb(dc, imm) instr_create_0dst_1src((dc), OP_dmb, (imm))
+
+/** Immediate values for INSTR_CREATE_dmb(). */
```

Comment:
This looks like an exact copy of the enum in arm/.  To avoid duplicating code, please instead move it to a shared header.  Maybe instr_create_shared.h, or maybe a new ir/aarchxx/instr_create_aarchxx.h, or possibly instr.h or opnd.h.

Reply:
(Wondering what happens to the duplicated enum values in the Doxygen generated docs?  Is there a warning from Doxygen?)

Reply:
Got it. I will change the code accordingly. It will take a while on my end, but I will submit a version by the end of Wed. I didn't look at the generated Doxygen docs, let me check it out. I don't remember it warns, perhaps I didn't notice one back then.

Reply:
I have had a look of those aforementioned files. Thank you!
instr.h: defines architectural values WRT instruction encoding/decoding.
opnd.h: defines architectural values WRT operand and register encoding/decoding.
instr_create_shared.h: seems like defining architecturally independent macros and definitions.

For the DMB macro and its IMM definitions, I feel creating a new ir/aarchxx/instr_create_aarchxx.h is a way to go.

Because DMB's IMM enum doesn't share the exactly same semantics with the other arch (e.g. x86), so the instr_create_shared.h is ruled out;

instr.h and opnd.h are more related to architectural definitions, rather than instruction-level definitions. So I think I will go with creating a new ir/aarchxx/instr_create_aarchxx.h and move both ARM and AArch64's DMB macro and IMM enum to the new file. If no objection, then I will go with this path forward. Thank you.

Reply:
ir/aarchxx/instr_create_aarchxx.h seems good to me.

Reply:
I have created instr_create_aarchxx.h in the latest commit, but I am struggling with making it appear in the Doxygen htmls.
Do you know how to make instr_create_aarchxx.h show up in the doxygen updates? Thank you.

Reply:
It needs to be added to the file list in genapi.pl.